doc/architecture/frontend.md

```diff
@@ -0,0 +1,57 @@
+# Frontend Architecture
+
+## Structure
+
+- `__mocks__/`: Contains mocked modules for tests
+- `assets/`: Contains asset files like images
+- `bindings/`: Contains the generated API bindings
+- `components/`: Contains components that are shared across features
+- `config/`: Contains initialization code for the environment and libraries
+- `features/`: Contains the distinct features of the application
+  - `dark_mode/`: Contains the dark_mode feature (Switcher and store)
+  - `imprint_page/`: Contains the imprint page
+  - `landing_page/`: Contains the landing page
+  - `map_planning/`: Contains the map with layers
+  - `seeds/`: Contains the seed routes
+- `generated/`: Contains generated code, for now only docs
+- `hooks/`: Contains hooks that are shared across features
+- `icons/`: Contains svg icons
+- `routes/`: Contains the routes of the application
+- `styles/`: Contains the global and miscellaneous style sheets
+- `utils/`: Contains additional utility functions
+
+```mermaid
+graph LR;
+  features -->|Uses another feature's public API| f
+  features -->|Uses bindings for API calls| bindings
+  features -->|Uses shared components| components
+  features -->|Calls functions| config
+  features -->|Uses hooks| hooks
+  features -->|Calls functions| utils
+  features -->|Uses icon files| icons
+  features -->|Uses asset files| assets
+  features -->|Uses some miscellaneous styles| styles
+  routes --> |Renders route features| features
+    subgraph f[example feature]
+      direction LR
+      E[index] -->|Components scoped to a specific feature| A[components]
+      E -->|API request declarations and api hooks related to a specific feature| B[api]
```

Comment:
Please describe (probably already before): What exactly is `B[API]`? Where does it live?

```suggestion
      E -->|API request declarations and API hooks related to a specific feature| B[API]
```